solidity/src/MaybeMintERC721.sol

```diff
@@ -3,6 +3,7 @@ pragma solidity 0.8.24;
 import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {CadenceArchUtils} from "onflow/random-coin-toss/solidity/src/CadenceArchUtils.sol";
```

Comment:
Should we extract `CadenceArchUtils` into a stand-alone repo? It doesn't make sense to add random coin toss as a dependency just to use a utility contract

Reply:
Good callout. I'm wondering if we should have a solidity contracts repo similar to [openzeppelin-contracts](https://github.com/OpenZeppelin/openzeppelin-contracts) so consumers can easily install any Flow EVM contracts. What do you think of this approach?

Reply:
Yeah good idea. Maybe we can call it `flow-sol-utils` vs. `contracts` that sound much more library/dependency/not-for-everyone

Reply:
I'll kick that repo off today. The only contracts I can think of adding at the moment are [`CadenceArchUtils.sol`](https://github.com/onflow/random-coin-toss/blob/main/solidity/src/CadenceArchUtils.sol) (library), [`CadenceRancomConsumer.sol`](https://github.com/onflow/random-coin-toss/blob/main/solidity/src/CadenceRandomConsumer.sol) (abstract contract) and [`Xorshift128plus.sol`](https://github.com/onflow/random-coin-toss/blob/main/solidity/src/Xorshift128plus.sol) (library). I'm sure we'll add more over time.

Actually, what do you think about adding the WFLOW contract into that repo?

Reply:
Yeah we can add WFLOW too

Reply:
Created the [onflow/flow-sol-utils](https://github.com/onflow/flow-sol-utils) repo to store these dependencies. Will update this repo to pull deps from there.

Reply:
Updated dependencies to pull from the created repo